pages/logout.tsx

```diff
@@ -1,40 +1,174 @@
+import axios from 'axios';
+import { GetServerSidePropsContext } from 'next';
 import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { FormEvent, useEffect, useState } from 'react';
+import { validateRequest } from '~/backend/auth';
+import { createFeedbackSession } from '~/backend/feedback';
 import { Button } from '~/components/Button';
 import { Logo } from '~/components/Logo';
 import { TitleText } from '~/components/TitleText';
+import { CreateFeedback } from '~/shared/types';
+import { handleGeneralError } from '~/utils/handleError';
+import Cookies from 'js-cookie';
+
+const POSSIBLE_ERRORS = {
+  'feedback was invalid!': 'Palauteteksti on virheellinen',
+  'uuid was invalid!':
+    'Yksilöintitunnus on virheellinen. Kokeile lähettää palaute uudelleen myöhemmin. Pahoittelemme tapahtunutta.',
+  'feedback text is mandatory!': 'Palauteteksti on pakollinen',
+  'server error!': 'Palvelin virhe',
+  'palvelin virhe!': 'Palvelin virhe',
+  'odottamaton virhe tapahtui!': 'Odottamaton virhe tapahtui',
+} as const;
+
+type KnownFrontEndErrorTexts = keyof typeof POSSIBLE_ERRORS;
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const validatedUser = await validateRequest(context.req, context.res);
+  if (!validatedUser.user) {
+    return {
+      redirect: {
+        permanent: false,
+        destination: '/',
+      },
+    };
+  }
+  await createFeedbackSession(context, validatedUser.user);
+  return {
+    props: {},
+  };
+}
 
 export default function Logout() {
-  return (
-    <main className="h-screen w-full max-w-full ">
-      <div className="flex h-screen w-screen flex-col items-center">
-        <div className="max-w-80">
-          <Logo />
-          <div className="flex flex-col items-center pt-10">
-            <TitleText className="font-bold">Näkemiin!</TitleText>
-            <p className="pt-10 text-center text-sm text-gray-600 [overflow-wrap:anywhere]">
-              Oli ilo auttaa sinua lahjaideoiden kanssa, nähdään pian uudelleen!
-            </p>
+  const [feedbackText, setFeedbackText] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [isFeedbackSent, setIsFeedbackSent] = useState(false);
+
+  const router = useRouter();
+
+  useEffect(() => {
+    logoutUser().catch((e) => {
+      console.error(e);
+    });
+  }, []);
+
+  async function logoutUser() {
+    try {
+      await axios.post('/api/auth/logout');
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  function handleInternalError(e: unknown) {
+    console.error(e);
+    const errorMessage =
+      POSSIBLE_ERRORS[
+        handleGeneralError(e).toLowerCase() as KnownFrontEndErrorTexts
+      ] || 'Palauteteksti tai yksilöintitunnus on virheellinen';
+    setErrorText(errorMessage);
+  }
```

Comment:
`handleInternalError`in tarkoitus on `pages/logout.tsx`-tiedoston "omat" virheet. Eli listalta hard-koodatut virhetekstit tai sitten fallback-teksti